scripts/runtests.py

```diff
@@ -0,0 +1,128 @@
+import json
+import re
+import subprocess
+import sys
+from pathlib import Path
+from typing import Optional
+
+try:
+    from termcolor import colored
+except ImportError:
+
+    def colored(text: str, color: str = "") -> str:  # type: ignore[misc]
+        return text
+
+
+# Popular Flake8 extensions not used in Typeshed that cause errors
+_STRICTER_CONFIG_FILE = "pyrightconfig.stricter.json"
+_SUCCESS = colored("Success", "green")
+_SKIPPED = colored("Skipped", "yellow")
+_FAILED = colored("Failed", "red")
+
+
+def _parse_jsonc(json_text: str) -> str:
+    # strip comments from the file
+    lines = [line for line in json_text.split("\n") if not line.strip().startswith("//")]
+    # strip trailing commas from the file
+    valid_json = re.sub(r",(\s*?[\}\]])", r"\1", "\n".join(lines))
+    return valid_json
+
+
+def _get_strict_params(stub_path: str) -> list[str]:
+    with open(_STRICTER_CONFIG_FILE) as file:
+        data = json.loads(_parse_jsonc(file.read()))
+    if stub_path in data["exclude"]:
+        return []
+    return ["-p", _STRICTER_CONFIG_FILE]
+
+
+if __name__ == "__main__":
+    try:
+        path = sys.argv[1]
```

Comment:
In future PRs, we could consider adding more command-line arguments. It might be nice to have an argument to run *all the tests*, not just the tests for a specific stubs directory. That would have been useful for a PR like #8796, which involved changes across a wide swathe of typeshed.

But I think we should keep this PR simple, and not do that for now :)